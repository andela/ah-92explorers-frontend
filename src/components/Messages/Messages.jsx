import React from 'react';

const Messages = ({ success, error, file }) => {
  switch (true) {
    case success !== undefined:
      return (
        <div className="successBox">
          { success }
        </div>
      );
    case error !== undefined:
      return (
        <div className="errorBox">
          { error }
        </div>
      );
    case file !== undefined:
      return (
        <div className="errorBox">
          { file }
        </div>
      );
    default:
      return (
        <div style={{ display: 'none' }} className="Nothing">
          Nothing
        </div>
      );
  }
};

export default Messages;
