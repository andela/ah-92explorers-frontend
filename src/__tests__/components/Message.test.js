import React from 'react';
import { shallow } from 'enzyme';
import Messages from '../../components/Messages/Messages.jsx';

describe('Renders CreateArticleComponent', () => {
  const props = {
    success: undefined,
    error: undefined,
    file: undefined
  };

  const successProps = {
    success: 'Mars',
    error: undefined,
    file: undefined
  };

  const errorProps = {
    success: undefined,
    error: 'undefined',
    file: undefined
  };

  const fileProps = {
    success: undefined,
    error: undefined,
    file: 'undefined'
  };

  it('should render messages component', () => {
    let wrapper = shallow(
        <Messages {...props} />,
      );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render a success component', () => {
    let wrapper = shallow(
        <Messages {...successProps} />,
      );
    expect(wrapper.find('.successBox').length).toBe(1);
  });
  it('should render a error component', () => {
    let wrapper = shallow(
        <Messages {...errorProps} />,
      );
    expect(wrapper.find('.errorBox').length).toBe(1);
  });
  it('should render a file component', () => {
    let wrapper = shallow(
        <Messages {...fileProps} />,
      );
    expect(wrapper.find('.errorBox').length).toBe(1);
  });
});