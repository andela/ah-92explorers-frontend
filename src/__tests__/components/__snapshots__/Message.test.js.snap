// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders CreateArticleComponent should render messages component 1`] = `ShallowWrapper {}`;
